src/build.rs

```diff
@@ -429,6 +528,33 @@ extern fn progress_cb(path: *const c_char,
     });
 }
 
+extern fn notify_cb(why: raw::git_checkout_notify_t,
+                    path: *const c_char,
+                    baseline: *const raw::git_diff_file,
+                    target: *const raw::git_diff_file,
+                    workdir: *const raw::git_diff_file,
+                    data: *mut c_void) -> c_int {
+    // pack callback etc
+    panic::wrap(|| unsafe {
+        let payload = &mut *(data as *mut CheckoutBuilder);
+        let callback = match payload.notify {
+            Some(ref mut c) => c,
+            None => return 0,
+        };
+        let path = if path.is_null() {
+            None
+        } else {
+            Some(util::bytes2path(CStr::from_ptr(path).to_bytes()))
+        };
+
+        callback(CheckoutNotificationType::from_bits_truncate(why),
+                 path,
+                 DiffFile::from_raw(baseline),
+                 DiffFile::from_raw(target),
+                 DiffFile::from_raw(workdir))
+    }).unwrap()
```

Comment:
Actually I think we shouldn't panic here, but I'm not sure. I think `panic::wrap()` only yields `None` if the contents panicked? A notification callback cancels the checkout if the callback returns a non-zero value. So I'm not sure if the contents panicking would be reason to merely cancel the checkout or if it's much more serious and grounds to panic outright.

So is the fact that `panic::wrap()`'s contents panics serious enough to propagate that panic, or should it only cancel the checkout?

If simply cancelling the checkout is fine, I'll replace this with `.unwrap_or(1)`, unless there's something else that would be preferable.